plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/SMTPSessionParametersCustomizer.java

```diff
@@ -53,15 +51,15 @@ public SMTPSessionParametersCustomizer(final TrustStore trustStore) {
 
   //@Override
   public Properties customize(final Properties params) {
-    final SSLContext sslContext = trustStore.getSSLContextFor(smtpTrustStoreKey());
     final String host = params.getProperty("mail.smtp.host");
     final String port = params.getProperty("mail.smtp.port");
-    if (sslContext != null && params.remove("mail.smtp.socketFactory.class") != null) {
+    if (Boolean.valueOf(params.getProperty("mail.smtp.ssl.useTrustStore"))
```

Comment:
I assume `mail.smtp.ssl.useTrustStore` is a nexus-specific property? (I don't see it listed in the table from https://javamail.java.net/nonav/docs/api/com/sun/mail/smtp/package-summary.html) If so then we should document this somewhere.